### ISSUE TITLE
A "few buttons" controller

### ISSUE THREAD

**Issue by harbaum** [vossstef/A2600Nano#7]
This needs a keyboard but only for very few functions. I've been told that these

https://www.atari-forum.com/download/file.php?id=53913&mode=view

cheap keyboard like things in fact do work with your core and can be configured with https://github.com/kriomant/ch57x-keyboard-tool

**Comment by vossstef** [comment]
@harbaum do we need a variant that covers all the OSD Key's as well (up, down, ok, page up, page down, F12 + Game Button's ) or will e.g. three Key's + Dial be sufficient ?

**Comment by harbaum** [comment]
Whatever the user wants. It seems these are fully programmable, so there's no work for us to do.

I just mentioned it since it might be useful for your core as a full keyboard seems to be total overkill.

**Comment by ojaksch** [comment]
May I join the discussion? I'm the one who is using this macro keyboard and pointed that to Till. In my case it is the 12-key/2-knob one, has three programmable layers (ie for different cores) and is freely programmable (the knobs also). I set it up with 10 keys for the A2600 (F11, F12 etc) and is doing a great job.

If there are any questions or things to try, just ask. One idea I'm having is to use one of the knobs as a mouse or a paddle emulation.

**Comment by vossstef** [comment]
@ojaksch , i also ordered the same 12 key + 2 knob but didn't found time to try it apart from checking USB ID and function in windows... 
Do you want to share your best practice mapping file?

**Comment by ojaksch** [comment]
Of course! See my attached mapping file and how rotation and mapping goes according to the yaml.
"F24" is meant for "unoccupied", haven't tried how to set a key to "no function".

[a2600-ch57x-mapping.zip](https://github.com/user-attachments/files/17742550/a2600-ch57x-mapping.zip)

![keyboard-12-2](https://github.com/user-attachments/assets/7f1f0f30-6425-477b-b512-62b1dae3a4c7)

**Comment by ojaksch** [comment]
Well, the next two test candidates have arrived and are again working as expected. Picture shows their key assignment, USB connector to north.
yaml's attached. The 9-key needed its orientation set to _clockwise_ to keep its USB connector to north. Otherwise nothing special to report, except having fun :smile: 

[a2600-ch57x-mappings.zip](https://github.com/user-attachments/files/17776976/a2600-ch57x-mappings.zip)

![IMG_20241115_153134](https://github.com/user-attachments/assets/1adf73dd-3c82-4834-910c-39703c58df35)

**Comment by ojaksch** [comment]
A nice extra to these keyboards with knobs would be if we (the FPGA Companion?) could use the knobs as a Paddle (simulated per mouse wheel) plus Trigger. Any normal wheel mouse could also be used and the ch57x-keyboard-tool already supports "wheelup", "wheeldown" and three mouse clicks (left, middle, right) - additionally with a modifier key like "ctrl-wheeldown", ctrl-click etc.

**Comment by harbaum** [comment]
Yeah. I ordered one many weeks ago. But it never arrived. That's actually the first time that a delivery from Aliexpress didn't arrive ...

I may order a second one.

**Comment by venice1200** [comment]
I got mine (4x3 keys & 2 knobs) today, works fine with the config from above.

**Comment by cebelcher** [comment]
I spent a little time today using dye sublimation to print titles on my keypad.  I wanted the arrows to be more orange like the Atari CX-40 joystick, but I'm pretty happy.

![Image](https://github.com/user-attachments/assets/bc38ae79-a4d2-42d6-89d7-ca20dc5f5c3f)
